airflow-core/src/airflow/serialization/serialized_objects.py

```diff
@@ -1690,7 +1691,7 @@ def serialize_dag(cls, dag: DAG) -> dict:
         except SerializationError:
             raise
         except Exception as e:
-            raise SerializationError(f"Failed to serialize DAG {dag.dag_id!r}: {e}")
+            raise SerializationError(f"Failed to serialize DAG {dag.dag_id!r}") from e
```

Comment:
Returning this to the original because it has more info. Maybe we can change this in another PR